test/parallel/test-cluster-worker-wait-server-close.js

```diff
@@ -22,10 +22,12 @@ if (cluster.isWorker) {
   worker.once('listening', function() {
     net.createConnection(common.PORT, common.localhostIPv4, function() {
       var socket = this;
-      worker.disconnect();
       setTimeout(function() {
-        socket.write('.');
-        connectionDone = true;
+        worker.disconnect();
```

Comment:
basically have it how it was originally except put the call into the `.disconnect()`. if that callback is never called then there must be an issue with the patch itself on some platforms.